src/plugins/skills/src/systems/skill_path_to_handle.rs

```diff
@@ -0,0 +1,297 @@
+use crate::{resources::SkillFolder, skills::Skill, traits::TryMap};
+use bevy::{
+	asset::{Asset, AssetEvent, AssetId, Assets, Handle},
+	ecs::{system::Res, world::Ref},
+	prelude::{Commands, Component, DetectChanges, Entity, EventReader, Query},
+};
+use common::{
+	errors::{Error, Level},
+	traits::{
+		get_handle_from_path::GetHandelFromPath,
+		load_asset::Path,
+		try_insert_on::TryInsertOn,
+	},
+};
+
+pub(crate) fn skill_path_to_handle<
+	TSource: Component + TryMap<Path, Handle<Skill>, TResult>,
+	TResult: Component,
+	TFolder: GetHandelFromPath<Skill> + Asset,
+>(
+	mut commands: Commands,
+	mut events: EventReader<AssetEvent<TFolder>>,
+	skill_folder: Res<SkillFolder<TFolder>>,
+	loaded_folders: Res<Assets<TFolder>>,
+	sources: Query<(Entity, Ref<TSource>)>,
+) -> Vec<Result<(), Error>> {
+	let folder_id = AssetId::from(skill_folder.0.clone());
+	let Some(folder) = loaded_folders.get(folder_id) else {
+		return vec![Err(no_skill_folder_error())];
+	};
+	let force_update = events.read().any(added_or_changed(folder_id));
+	let mut errors = vec![];
+
+	for (entity, source) in sources.iter().filter(is_changed_or(force_update)) {
```

Comment:
remove source to reduce component clutter

Reply:
Removal breaks breaks functionality. For instance: remap paths, once folder assets change.

Reply:
This argument holds for handle -> struct. But letting initial path definitions potentially override changes in what handles are used might bring the game in an inconsistent state.

For instance, this should never happen:
- use skill with path: "my/path/a"
- later swap item with item from inventory, which uses skill on path "my/path/b"
- for some reason skill folder changes
- now override swapped out item (using skill "b") with the "original" skill, which is skill "a"

Reply:
Resolved: Needed a rework of slot initialization (the empty slot component was removed too early).